airflow/providers/google/cloud/example_dags/example_cloud_memorystore.py

```diff
@@ -161,7 +161,7 @@
     set_acl_permission = GCSBucketCreateAclEntryOperator(
         task_id="gcs-set-acl-permission",
         bucket=BUCKET_NAME,
-        entity="user-{{ task_instance.xcom_pull('get-instance')['persistenceIamIdentity']"
+        entity="user-{{ task_instance.xcom_pull('get-instance')['persistence_iam_identity']"
```

Comment:
Does this suggest a breaking change?

Reply:
Yes. Unfortunately yes. Upgrading the library version introduces minor behavioral differences, so we will have to add an update entry for these libraries in UPDATING.md.

Reply:
CAan we keep backwards-compatibility? I think we should change our approach and only introduce backwards-incompatible changes in providers if we absolutely must and there is no other way to solve it.

Reply:
This will be very problematic because this is changed in the third library and we are just passing that object as JSON. These are the consequences of the fact that our operators are only wrappers on the API and the abstraction layer between the third library and the DAG is relatively thin.

In my opinion, we need to introduce a breaking change. Otherwise, we will be forced to use old and outdated libraries that will not allow us to use new API fields/features.

Reply:
I added docs.

Reply:
Is the list of those modified fields known and fixed ? I guess it should be as it comes from the old version of library which is not going to be updated. I guess the values in Xcom come from another operator of ours, and they contain the values returned by `CloudMemorystoreGetInstanceOperator` - result of `MessageToDict` method call?

I guess in this case we could -  potentially make it backwards-compatible by modifying the dictionary returned and copying the values with legacy field names?  How feasible it is ? Maybe in the new library there is already a fixed list of values or even a code that we could use to do it ?

Reply:
The word list is not provided explicitly, but we may try to guess these names when comparing types with similar names in the new version of the library, but this will require creating a large key mapping from one version of the library to the new library.  This key list is also not fixed as new versions of libraries may contain new keys. 

This has its limitation. If we only replace dict keys based on a predefined list of keys, we can accidentally modify user values as well, because Google libraries not often, but sometimes also store values from the user as a key.
https://github.com/googleapis/googleapis/blob/eabe7c0fde64b1451df6ea171b2009238b0df07c/google/cloud/datacatalog/v1/tags.proto#L79
https://github.com/googleapis/googleapis/blob/eabe7c0fde64b1451df6ea171b2009238b0df07c/google/cloud/redis/v1/cloud_redis.proto#L276
For this not to happen, we may collect such cases as path, but we will still have to maintain them. 

PS. Google uses the snake_case convention in protobuff, so with this change, Google tried to standardize all field names.

Reply:
There is one more problem. If we try to rename the keys, then it will not be possible to create an object using this dictionary.
```
instance_dict = t.xcom_pull('get-instance')
instance = Instance(**instance_dict)
```

Reply:
In this case, I think we should put that change on hold and release it together with other breaking changes for google provider. I do not think we have an immediate need to migrate to this new '2.0.0' version. And this should happen IMHO at the point we add more breaking changes together.

Reply:
All right - after the discussion - seems like we have more breaking changes in this "batch" so this will be fine to make it backwards incompatible and set version to 2.0.0